token-syncer/src/token_syncer/commands/syncer.clj

```diff
@@ -78,75 +78,71 @@
 
-                  (and latest-root
-                       (= latest-token-description description))
-                  {:code :success/token-match}
+                      (and latest-root
+                           (= latest-token-description description))
+                      {:code :success/token-match}
 
-                  ;; deleted on both clusters, irrespective of remaining values
-                  (and (get latest-token-description "deleted")
-                       (get description "deleted"))
-                  {:code :error/tokens-deleted
-                   :details {:message "soft-deleted tokens should have already been hard-deleted"}}
+                      ;; deleted on both clusters, irrespective of remaining values
+                      (and (get latest-token-description "deleted")
+                           (get description "deleted"))
+                      {:code :error/tokens-deleted
+                       :details {:message "soft-deleted tokens should have already been hard-deleted"}}
 
-                  (and (some? opt-out-metadata-name)
-                       ;; opt-out of syncing if any of the descriptions signal opt-out intent
-                       (or (= "true" (get-in latest-token-description ["metadata" opt-out-metadata-name]))
-                           (= "true" (get-in description ["metadata" opt-out-metadata-name]))))
-                  {:code :success/skip-opt-out}
+                      (and (some? opt-out-metadata-name)
+                           ;; opt-out of syncing if any of the descriptions signal opt-out intent
+                           (or (= "true" (get-in latest-token-description ["metadata" opt-out-metadata-name]))
+                               (= "true" (get-in description ["metadata" opt-out-metadata-name]))))
+                      {:code :success/skip-opt-out}
 
-                  ;; active token, content the same irrespective of system metadata keys but roots different
-                  (and (seq description)
-                       (not= latest-root cluster-root)
-                       (not (get latest-token-description "deleted"))
-                       (not (get description "deleted"))
-                       (= (apply dissoc latest-token-description system-metadata-keys)
-                          (apply dissoc description system-metadata-keys)))
-                  (if (= latest-update-user cluster-update-user)
-                    {:code :success/skip-token-sync}
-                    {:code :error/token-sync
-                     :details {:message "token contents match, but were edited by different users"}})
+                      ;; token user-specified content (including "deleted") is different, and the last update user or root is different
+                      (and (seq description)
+                           (not= latest-root cluster-root)
+                           (not= latest-update-user cluster-update-user)
```

Comment:
We are okay if `root` and `cluster` do not match now as long as rest of the content matches. Should the two lines above form an `or` clause?

Reply:
This is a different situation where the contents don't match. Later in the `and` call:
```
(not= (apply dissoc description ignored-root-mismatch-equality-comparison-keys)
                                 (apply dissoc latest-token-description ignored-root-mismatch-equality-comparison-keys))
```

Reply:
Oops, my bad I read one of the terms as dealing with `cluster` instead of `last-update-user`